libs/prisma-models/src/field/relation.rs

```diff
@@ -127,7 +127,9 @@ impl RelationField {
     /// The set of fields required by the relation (on the model of the relation field) to be able to link the related records.
     /// Tbd examples
     pub fn linking_fields(&self) -> ModelIdentifier {
-        if self.relation_info.to_fields.is_empty() {
+        if self.relation().is_many_to_many() {
+            self.model().primary_identifier()
```

Comment:
The `to_fields` is populated on both sides of a relation for many 2 many. I think this caused problems because of the underlying datasource fields.